### PR TITLE
fix: Auto height fixes for Auto Layout

### LINKED ISSUE #21300: [Bug]: [Auto layout] Tab widget jumps while switching from one tab to another
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When we drag and drop a tab widget and dnd some widgets inside the tab widget in both tab, now switch between tabs, we are able to see the a jump.

https://user-images.githubusercontent.com/104895602/224015059-70ee6a89-6c30-470f-b292-96962c12adcd.mp4

### Steps To Reproduce

1. Dnd a tab widget
2. Dnd some widgets in both tabs
3. Switch between both tabs
4. Observe a jump

### Public Sample App

_No response_

### Environment

Deploy Preview

### Issue video log

_No response_

### Version

Appsmith Community v1.9.11-SNAPSHOT

### LINKED ISSUE #21758: [Bug]: [Widget resizing] Nested Group widget resizing isnt smooth
Step 1: Dnd a container widget
Step 2: Dnd a fill widget and hug widget in same line (Check box group, Swtich group, Radio group)
Step 3: Try to expand the hug widget
Step 4: Observe the fill widget is taking the whole space while we resize the hug widget
Step 5: Once we release we are able to see the hug widget is expanded
Step 6: Same happens when we reduce the size of hug widget
Step 7: If we add more options in the hug widgets and resize, the resize isnt done properly
Step 8: Check zoom for more clarity on the issue.

https://user-images.githubusercontent.com/104895602/227550082-17fb312e-7d65-4820-bc2b-8f8466471757.mp4

### LINKED ISSUE #21837: [Bug] Modal widgets when resized to mobile viewport, having issues with the auto height
https://www.loom.com/share/b67ce0e6b6444900862e1fa28abdf079

### LINKED ISSUE #21870: [Bug[Widget Responsiveness] When resized to mobile viewport(using resize handles and actual viewport), some of the widgets in List widget are not responding correctly from second item container onwards
1. Mobile viewport using handles
https://www.loom.com/share/0d2db381fb3441d2a151efa01bfa44e2

2. Actual mobile viewport

Screenshot 2023-03-29 at 5 06 57 PM

### LINKED ISSUE #21977: [Bug]: Container's height is not getting reduced
https://user-images.githubusercontent.com/10436935/229191788-d522f56f-61a2-4c76-8c0c-228f33b62c7a.mov

### LINKED ISSUE #22086: [Bug][Widget Responsiveness] the child widgets of list widget are not visible in actual mobile viewport
Image

### LINKED ISSUE #22093: [Bug][Widget Responsiveness] the space between child widgets are not static in different viewports for List widget _autolayout mode
https://www.loom.com/share/f7448fc9c5fe4162be6d7a11858fc4e0

### LINKED ISSUE #22183: [Bug]: Resize loop
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Resize loop due to autoLayoutDimensionObserver, and bounding box computation

https://user-images.githubusercontent.com/103687/230609541-5f228850-e990-400f-a365-f850a8c20f63.mov

### Steps To Reproduce

1. Add a few widgets include input widgets in desktop viewport
2. Resize quickly to lower than mobile viewport

### Public Sample App

_No response_

### Environment

Deploy Preview

### Issue video log

_No response_

### Version

Deploy preview

### LINKED ISSUE #22400: [Bug]:[Widget responsiveness] When we drop RTE resize handler is moved without even click and drag of mouse
Drag and drop a RTE
Click on the Resize handler and release the click of mouse
Move the mouse besides the resize handler
Observe the viewport is resize without the intervention of mouse click
We are able to release this action when we click out side of the canvas

https://user-images.githubusercontent.com/104895602/231874039-d78ea249-4ccf-439a-86fa-57e9fba2df1c.mp4

### LINKED ISSUE #22539: [Bug][Widget responsiveness]Issue List widget in mobile viewport
When we Dnd list widget in Auto layout and change canvas to mobile viewport, we are able to see 1st option is different from the other options below

### LINKED ISSUE #22541: [Bug]: [Widget responsiveness] Resize handler is non deselectable
https://user-images.githubusercontent.com/104895602/232978476-47d2d7de-ad8e-41a7-9607-95ea77ec99d7.mp4

### PR DESCRIPTION
**Pull request by marks0351**
> Pull Request Template
>
> Use this template to quickly create a well written pull request. Delete all quotes before creating the pull request.

### Auto Height flow in Auto Layout Editor.
Problem: Usage of fixed layout editor's Auto Height was leading to several bugs because of the difference in nature of both editors. and patching it to cater to use cases of auto layout made little sense and also increases the risk of maintaining it. So we have essentially fixed our position utils to cater to all uses cases that Auto height was dealing with.

### List widget related issues in Auto Layout editor
In the case of widgets that are managed by List widget (Meta widgets), we store it in `entitites.metaWidgets`.
For auto layout calculations, we were making use of `entities.canvasWidgets` and not `entites.metaWidgets` caused the widgets inside List widget not to function properly.
In this PR, we started making use of both canvasWidgets & metaWidgets to resolve some of the issues with ListWidget in Auto layout mode.

This PR fixes a bunch of Auto Height bugs, List widget bugs as well as a polish related issues.

Fixes #21977
Fixes #22093
Fixes #21837
Fixes #21300
Fixes #22183
Fixes #21758
Fixes #22400
Fixes #22541
Fixes #21870
Fixes #22086
Fixes #22539

> if no issue exists, please create an issue and ask the maintainers about this first

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update

- Manual
- Jest
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/components/designSystems/appsmith/autoLayout/FlexComponent.tsx [hunk 1]
```diff
@@ -99,11 +99,12 @@ export function FlexComponent(props: AutoLayoutProps) {
     width: isResizing
       ? "auto"
       : `${props.componentWidth - WIDGET_PADDING * 2 + RESIZE_BORDER_BUFFER}px`,
-    height: isResizing
-      ? "auto"
-      : `${
-          props.componentHeight - WIDGET_PADDING * 2 + RESIZE_BORDER_BUFFER
-        }px`,
+    height:
+      isResizing || props.isCanvas
```
Comment (prsidhu) [inline app/client/src/components/designSystems/appsmith/autoLayout/FlexComponent.tsx:103]: why is this needed?

File: app/client/src/widgets/ContainerWidget/component/index.tsx [hunk 2]
```diff
@@ -155,7 +155,10 @@ function ContainerComponent(props: ContainerComponentProps) {
         onClick={props.onClick}
         onClickCapture={props.onClickCapture}
         resizeDisabled={props.resizeDisabled}
-        shouldScrollContents={props.shouldScrollContents}
+        shouldScrollContents={
```
Comment (aswathkk) [inline app/client/src/widgets/ContainerWidget/component/index.tsx:158]: @marks0351 Is this the right way to do this?
Can we set the value of `shouldScrollContents` to false in withWidgetProps if it's autoLayout?
